builtin/src/genesis.rs

```diff
@@ -177,7 +179,7 @@ pub fn create_genesis_actors<B: WorkbenchBuilder>(
     )?;
 
     // Faucet account
-    let faucet_id = BURNT_FUNDS_ACTOR_ID - 1;
+    let faucet_id = TEST_FAUCET_ADDR.id().unwrap();
```

Comment:
@anorth Changing this from `BURNT_FUNDS_ACTOR_ID - 1` (98) to `TEST_FAUCET_ADDR` (102) breaks the `create_accounts` call for some reason. 

The FVM itself sees a zero balance for the faucet_addr and the transfer during the send fails with "insufficient funds".  Even though if I call `v.actor(TEST_FAUCET_ADDR).balance` to load the state directly before the send, I see the correct balance. 

I'm not sure what is going on here. If I patch locally to builtin-actors and modify its `create_accounts` to use the ID 98 instead, it all seems to work. Is there some fundamental difference between an actor using a ID below or above FIRST_NON_SINGLETON_ID.

Reply:
https://github.com/anorth/fvm-workbench/pull/23 demonstrates the "fix"

Reply:
I'm sure there's a good explanation, but I don't know what it is. There's nothing fundamentally different about actor IDs <100, I don't think the VM handles them any differently. There might be hardcoded assumptions somewhere (in actors) that we're missing.

Reply:
InitActor was being created with a default state. So it was allocating IDs to new actors that overlapped with the test_faucet_address (and anything else seeded as a part of genesis state)